library/system/src/lib/yast2/ui_plugin_info.rb

```diff
@@ -0,0 +1,162 @@
+# ------------------------------------------------------------------------------
+# Copyright (c) 2021 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of version 2 of the GNU General Public License as published by the
+# Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+#
+# ------------------------------------------------------------------------------
+
+require "yast"
+require "yast2/shared_lib_info"
+
+module Yast
+  # Class to get information about UI plug-ins used by a process (by default
+  # the current process) from its /proc/self/maps file.
+  #
+  # You can also get information for a different process by specifying its
+  # /proc/$pid/maps file.
+  #
+  # For testing, a fixed file can be used.
+  #
+  # The information stored in this class is only a snapshot in the life time of
+  # the process. As new shared libs are loaded (e.g. plug-ins are loaded with
+  # dlopen()), this information may become outdated. In that case, simply let
+  # the old instance go out of scope and create a new one.
+  #
+  # More information: https://github.com/yast/yast-yast2/pull/1194
+  #
+  class UiPluginInfo < SharedLibInfo
```

Comment:
Maybe I'd use the `UIPluginInfo` name as `UI` is an abbreviation (similar to `URI` std Ruby class).

Reply:
I wasn't sure about that. We tend to be _very_ consistent about those CamelCase names in other places. I have no strong opinion one way or the other.

What do others think?

Reply:
@jreidinger, @dgdavid, @ancorgs  also voted for renaming it to `UIPluginInfo`. Done.